ttl/ttl_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2024 Hemi Labs, Inc.
+// Use of this source code is governed by the MIT License,
+// which can be found in the LICENSE file.
+
+package ttl
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+)
+
+func callback(key any, value any) {
+	v, ok := value.(*sync.WaitGroup)
+	if !ok {
+		panic(fmt.Sprintf("invalid value type: %T", value))
+	}
+	v.Done()
+}
+
+func callbackPanic(key any, value any) {
+	panic(fmt.Sprintf("unexpected callback: %v", spew.Sdump(key)))
+}
+
+func TestTTLExpire(t *testing.T) {
+	count := 10
+	tm, err := New(count)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var wg sync.WaitGroup
+	for i := 0; i < count; i++ {
+		wg.Add(1)
+		tm.Put(ctx, time.Second, strconv.Itoa(i), &wg, callback, nil)
+	}
+	for i := 0; i < count; i++ {
+		key := strconv.Itoa(i)
+		_, err := tm.Get(key)
+		if err != nil {
```

Comment:
nit: I think this error check could be inlined:
```suggestion
		if _, err := tm.Get(key); err != nil {
```